addons/crm/models/crm_lead.py

```diff
@@ -0,0 +1,1188 @@
+
+    color = fields.Integer(string='Color Index', default=0)
+    partner_address_name = fields.Char(related='partner_id.name', string='Partner Contact Name', readonly=True)
+    partner_address_email = fields.Char(related='partner_id.email', string='Partner Contact Email', readonly=True)
+    company_currency = fields.Many2one(related='company_id.currency_id', string='Currency', readonly=True, relation="res.currency")
+    user_email = fields.Char(related='user_id.email', string='User Email', readonly=True)
+    user_login = fields.Char(related='user_id.login', string='User Login', readonly=True)
+
+    # Fields for address, due to separation from crm and res.partner
+    street = fields.Char()
+    street2 = fields.Char()
+    zip = fields.Char(change_default=True, size=24)
+    city = fields.Char()
+    state_id = fields.Many2one("res.country.state", string='State')
+    country_id = fields.Many2one('res.country', string='Country')
+    phone = fields.Char()
+    fax = fields.Char()
+    mobile = fields.Char()
+    function = fields.Char()
+    title = fields.Many2one('res.partner.title')
+    company_id = fields.Many2one('res.company', string='Company', index=True, default=lambda self: self.env.user.company_id.id)
+    meeting_count = fields.Integer(compute='_compute_meeting_count', string='# Meetings')
+    lost_reason = fields.Many2one('crm.lost.reason', string='Lost Reason', index=True, track_visibility='onchange')
+
+    _sql_constraints = [
+        ('check_probability', 'check(probability >= 0 and probability <= 100)', 'The probability of closing the deal should be between 0% and 100%!')
+    ]
+
+    @api.depends('date_open', 'date_closed')
+    def _compute_day(self):
+        """
+        :return dict: difference between current date and log date
+        """
+        for lead in self:
+            if lead.date_open:
+                date_create = fields.Datetime.from_string(lead.create_date)
+                date_open = fields.Datetime.from_string(lead.date_open)
+                ans = date_open - date_create
+                lead.day_open = abs(int(ans.days))
+
+            if lead.date_closed:
+                date_create = fields.Datetime.from_string(lead.create_date)
+                date_close = fields.Datetime.from_string(lead.date_closed)
+                ans = date_close - date_create
+                lead.day_close = abs(int(ans.days))
+
+    @api.one
+    def _compute_meeting_count(self):
+        self.meeting_count = self.env['calendar.event'].search_count([('opportunity_id', '=', self.id)])
+
+    @api.onchange('partner_id')
+    def on_change_partner_id(self):
+        partner = self.partner_id
+        self.partner_name = (partner.parent_id and partner.parent_id.name) or (partner.is_company and partner.name) or False
+        self.contact_name = (not partner.is_company and partner.name) or False
+        self.title = partner.title and partner.title.id or False
```

Comment:
simply self.title = partner.title.id. partner.title is many2one field return recordset(s). no need to set False